src/zeek.bif

```diff
@@ -3175,14 +3175,16 @@ function bytestring_to_hexstr%(bytestring: string%): string
 	%{
 	zeek_uint_t len = bytestring->AsString()->Len();
 	const u_char* bytes = bytestring->AsString()->Bytes();
-	auto hextr_buf = std::make_unique<char[]>((2 * len) + 1);
-	auto hexstr = hextr_buf.get();
+	auto hexstr_buf = std::make_unique<char[]>((2 * len) + 1);
+	auto hexstr = hexstr_buf.get();
 
-	hexstr[0] = 0;
 	for ( zeek_uint_t i = 0; i < len; ++i )
-		snprintf(hexstr + (2 * i), 3, "%.2hhx", bytes[i]);
+		zeek::util::bytetohex(bytes[i], &hexstr[2 * i]);
+	hexstr[2 * len] = 0;
 
-	return zeek::make_intrusive<zeek::StringVal>(hexstr);
+	// Adopt the memory...
+	String* s = new String(1, reinterpret_cast<uint8_t*>(hexstr_buf.release()), 2 * len);
```

Comment:
```suggestion
	String* s = new String(true, reinterpret_cast<uint8_t*>(hexstr_buf.release()), 2 * len);
```

Reply:
Done, and also for the other occasions in Conn.cc